sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosPagedFluxTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ */
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.InternalObjectNode;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.rx.TestSuiteBase;
+import com.azure.cosmos.util.CosmosPagedFlux;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CosmosPagedFluxTest extends TestSuiteBase {
+
+    private static final int NUM_OF_ITEMS = 10;
+
+    private CosmosAsyncClient cosmosAsyncClient;
+    private CosmosAsyncContainer cosmosAsyncContainer;
+
+    @Factory(dataProvider = "clientBuilders")
```

Comment:
you will need client builder for session consistency otherwise the test will become flaky.

perhaps `clientBuildersWithDirectSession` ?